community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/PointFunction.scala

```diff
@@ -31,21 +31,24 @@ case class PointFunction(data: Expression) extends NullInNullOutExpression(data)
   override def compute(value: Any, ctx: ExecutionContext)(implicit state: QueryState): Any = value match {
     case IsMap(mapCreator) =>
       val map = mapCreator(state.query)
-      map.getOrElse("crs", CRS.WGS84.name) match {
-        case CRS.Cartesian.name =>
-          if (!map.contains("x") || !map.contains("y")) throw new InvalidArgumentException("A cartesian point must contain 'x' and 'y' coordinates")
-          val x = safeToDouble(map("x"))
-          val y = safeToDouble(map("y"))
-          CartesianPoint(x, y)
-
-        case CRS.WGS84.name =>
-          if (!map.contains("longitude") || !map.contains("latitude")) throw new SyntaxException("A cartesian point must contain 'x' and 'y' coordinates")
-          val longitude = safeToDouble(map("longitude"))
-          val latitude = safeToDouble(map("latitude"))
-          GeographicPoint(longitude, latitude, CRS.WGS84)
-
-        case unknown => throw new InvalidArgumentException(s"$unknown is not a supported coordinate system, supported values " +
-                                                    s"are ${CRS.Cartesian.name} and ${CRS.WGS84.name}")
+      if (map.contains("x") && map.contains("y")) {
+        val x = safeToDouble(map("x"))
+        val y = safeToDouble(map("y"))
+        val crsName = map.getOrElse("crs", CRS.Cartesian.name).asInstanceOf[String]
+        val crs = CRS.fromName(crsName)
+        crs match {
+          case CRS.WGS84 => GeographicPoint(x, y, crs)
+          case null => throw new InvalidArgumentException(s"'$crsName' is not a supported coordinate reference system for points, supported CRS are: '${CRS.WGS84.name}', '${CRS.Cartesian.name}'")
```

Comment:
`CRS.fromName` will not return `null` it will already have thrown an `InvalidArgumentException`?